cpp/test/core/blob_tests.cpp

```diff
@@ -33,6 +33,124 @@ TEST_CASE_BEGIN(LargeBlob)
 }
 TEST_CASE_END
 
+TEST_CASE_BEGIN(BlobConstruct)
+{
+    using bond::blob;
+
+    const uint32_t size = 32;
+
+    // test with shared_ptr
+    {
+        // move from other instance
+        blob a;
+        boost::shared_ptr<const char[]> data(new char[size], [](const char* data){delete[] data;});
+        blob b{data, size};
+
+        UT_AssertAreEqual(a.size(), 0);
+        UT_AssertAreEqual(b.size(), 32);
+        UT_AssertAreEqual(data.get(), b.data());
+        a = std::move(b);
+        // move guarantees b is still valid but may not have a value
+        UT_AssertAreEqual(a.data(), data.get());
+        UT_AssertAreEqual(b.data(), nullptr);
+        // data and a hold the shared_array, b no longer holds it
+        UT_AssertAreEqual(data.use_count(), 2);
+    }
+    {
```

Comment:
Can you simplify and have only below cases:
- default constructed blob,
- blob with shared memory,
- blob with raw memory.